src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp

```diff
@@ -39,165 +39,37 @@
 // systems using card-table-based write barriers, the efficiency of this
-// operation may be important.  Implementations of the "BlockOffsetArray"
+// operation may be important.  Implementations of the BlockOffsetTable
 // class may be useful in providing such efficient implementations.
-//
-// BlockOffsetTable (abstract)
-//   - BlockOffsetArray (abstract)
-//     - BlockOffsetArrayContigSpace
-//
 
-class ContiguousSpace;
-
-//////////////////////////////////////////////////////////////////////////
-// The BlockOffsetTable "interface"
-//////////////////////////////////////////////////////////////////////////
-class BlockOffsetTable {
-  friend class VMStructs;
-protected:
-  // These members describe the region covered by the table.
-
-  // The space this table is covering.
-  HeapWord* _bottom;    // == reserved.start
-  HeapWord* _end;       // End of currently allocated region.
-
-public:
-  // Initialize the table to cover the given space.
-  // The contents of the initial table are undefined.
-  BlockOffsetTable(HeapWord* bottom, HeapWord* end):
-    _bottom(bottom), _end(end) {
-    assert(_bottom <= _end, "arguments out of order");
-    assert(BOTConstants::card_size() == CardTable::card_size(), "sanity");
-  }
-
-  // Note that the committed size of the covered space may have changed,
-  // so the table size might also wish to change.
-  virtual void resize(size_t new_word_size) = 0;
-
-  virtual void set_bottom(HeapWord* new_bottom) {
-    assert(new_bottom <= _end, "new_bottom > _end");
-    _bottom = new_bottom;
-    resize(pointer_delta(_end, _bottom));
-  }
-
-  // Requires "addr" to be contained by a block, and returns the address of
-  // the start of that block.
-  virtual HeapWord* block_start_unsafe(const void* addr) const = 0;
-
-  // Returns the address of the start of the block containing "addr", or
-  // else "null" if it is covered by no block.
-  HeapWord* block_start(const void* addr) const;
-};
-
-//////////////////////////////////////////////////////////////////////////
-// One implementation of "BlockOffsetTable," the BlockOffsetArray,
-// divides the covered region into "N"-word subregions (where
-// "N" = 2^"LogN".  An array with an entry for each such subregion
-// indicates how far back one must go to find the start of the
-// chunk that includes the first word of the subregion.
-//
-// Each BlockOffsetArray is owned by a Space.  However, the actual array
-// may be shared by several BlockOffsetArrays; this is useful
-// when a single resizable area (such as a generation) is divided up into
-// several spaces in which contiguous allocation takes place.  (Consider,
-// for example, the garbage-first generation.)
-
-// Here is the shared array type.
-//////////////////////////////////////////////////////////////////////////
-// BlockOffsetSharedArray
-//////////////////////////////////////////////////////////////////////////
 class BlockOffsetSharedArray: public CHeapObj<mtGC> {
```

Comment:
Should be prefixed by `Serial`, same as `BlockOffsetTable` (since this is a complete reimplementation anyway).

Reply:
Maybe this renaming can be done in its own (trivial) PR for easier reviewing.

Reply:
sure np.